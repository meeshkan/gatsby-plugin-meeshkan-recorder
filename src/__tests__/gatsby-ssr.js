const { onRenderBody } = require(`../gatsby-ssr`);

describe(`gatsby-plugin-meeshkan-recorder`, () => {
	it(`injects a script tag if clientId is set`, async () => {
		const pluginOptions = {
			clientId: 'b00ecdd5-d1f7-4a87-9d1f-142cc16c78d8',
		};

		const setHeadComponents = jest.fn();

		await onRenderBody(
			{
				setHeadComponents,
			},
			pluginOptions
		);

		expect(setHeadComponents).toMatchSnapshot();
		expect(setHeadComponents).toHaveBeenCalledTimes(1);
	});

	it(`does not inject a script tag if clientId is not set`, async () => {
		const pluginOptions = {};
		const setHeadComponents = jest.fn();

		await onRenderBody(
			{
				setHeadComponents,
			},
			pluginOptions
		);

		expect(setHeadComponents).toMatchSnapshot();
		expect(setHeadComponents).toHaveBeenCalledTimes(0);
	});
});
