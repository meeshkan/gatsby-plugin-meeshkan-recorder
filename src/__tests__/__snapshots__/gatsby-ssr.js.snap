// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-meeshkan-recorder does not inject a script tag if clientId is not set 1`] = `[MockFunction]`;

exports[`gatsby-plugin-meeshkan-recorder injects a script tag if clientId is set 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Array [
        <script
          src="https://recorder.meeshkan.com/record.js?client_id=b00ecdd5-d1f7-4a87-9d1f-142cc16c78d8"
        />,
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
